support/jwt/src/main/kotlin/org/doorip/support/jwt/JwtProvider.kt

```
@@ -0,0 +1,87 @@
+        }
+    }
+
+    private fun parsePayload(token: String, key: SignatureKey): Claims? {
+        try {
+            val parser = when (val signatureKey = key.key) {
+                is SecretKey -> generateParser(signatureKey)
+                is PublicKey -> generateParser(signatureKey)
+                else -> return null
+            }
+            return parser.parseSignedClaims(token).payload
+        } catch (e: JwtException) {
```

detekt (1.23.7)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)